### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by JohnGrey17**
please check my solution

### CODE REVIEW

File: src/main/java/core/basesyntax/Color.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+public enum Color {
+
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Color.java:4]:
```suggestion
```

File: src/main/java/core/basesyntax/AreaCalculator.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package core.basesyntax;
+
+public interface AreaCalculator {
+
```
Comment (aaananas) [inline src/main/java/core/basesyntax/AreaCalculator.java:4]:
```suggestion
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+
```
Comment (aaananas) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]:
```suggestion
```

File: src/main/java/core/basesyntax/Figure.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public abstract class Figure implements AreaCalculator, Drawable {
+
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Figure.java:4]:
```suggestion
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
```
Comment (aaananas) [inline src/main/java/core/basesyntax/FigureSupplier.java:6]:
```suggestion
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            case 1:
+                return getRandomRectangle();
+            case 2:
+                return getRandomRightTriangle();
+            case 3:
+                return getRandomSquare();
+            case 4:
+                return getRandomIsoscelesTrapezoid();
+            default:
+                return getRandomFigure();
```
Comment (aaananas) [inline src/main/java/core/basesyntax/FigureSupplier.java:26]:
```suggestion
            case 4:
            default:
                return getRandomIsoscelesTrapezoid();
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        }
+    }
+
+    protected Circle getDefaultCircle() {
+        Circle circle = new Circle();
+        circle.setColor("White");
+        circle.setRadius(10);
```
Comment (aaananas) [inline src/main/java/core/basesyntax/FigureSupplier.java:33]: lets add constants for those fields

File: src/main/java/core/basesyntax/Circle.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    public int getRadius() {
+        return radius;
+    }
+
+    public void setRadius(int radius) {
+        this.radius = radius;
+    }
+
+    @Override
+        public void draw() {
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Circle.java:15]:
```suggestion
    public void draw() {
```

File: src/main/java/core/basesyntax/Circle.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+        System.out.println("Figure: Circle, " + " area:" + getRadius() + " sq.units, "
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Circle.java:16]:
```suggestion
        System.out.println("Figure: Circle, " + " area:" + getArea() + " sq.units, "

```

File: src/main/java/core/basesyntax/IsoscelesTrapezoid.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+
+    private int side1;
+    private int side2;
```
Comment (aaananas) [inline src/main/java/core/basesyntax/IsoscelesTrapezoid.java:6]: numbers in variable names are a bad practice

File: src/main/java/core/basesyntax/HelloWorld.java [hunk 11]
```diff
@@ -1,8 +1,10 @@
...
-
-}
+package core.basesyntax;
+
+/**
+ * Feel free to remove this class and create your own.
+ */
+public class HelloWorld {
+
+}
```
Comment (aaananas) [inline src/main/java/core/basesyntax/HelloWorld.java:8]: remove this class

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+        rectangle.getArea();
+        return rectangle;
+    }
+
+    protected Figure getRandomCircle() {
+        int count = random.nextInt(FIGURE_COUNT);
+        Circle circle = new Circle();
+        circle.setColor(colorSupplier.getRandomColor().name());
+        circle.setRadius(count);
+        circle.getArea();
```
Comment (aaananas) [inline src/main/java/core/basesyntax/FigureSupplier.java:71]: you dont need to call this method in methods here

**Review by aaananas — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/FigureSupplier.java (hunk 12 above)
Comment (JohnGrey17) [reply src/main/java/core/basesyntax/FigureSupplier.java:71]: do u mean all methods when i am trying to create random figure or just for Default Figure ?

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**